__tests__/app.test.js

```diff
@@ -420,4 +425,66 @@ describe('GET /api/articles', () => {
         expect(body.msg).toBe("Invalid order value");
       });
     });
-  });
\ No newline at end of file
+  });
+
+  //TASK 12 (which builds on tasks 5 & 11)
+  describe('GET /api/articles', () => {
+    test('200: serves an array of articles, filtered by topic', () => {
+      return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({body}) => {
+          const { articles } = body;
+          expect(Array.isArray(articles)).toBe(true);
+          expect(articles).toBeSortedBy("created_at", { descending: true });
+          articles.forEach((article) => {
+            expect(article).toEqual(
+              expect.objectContaining({
+                author: expect.any(String),
+                title: expect.any(String),
+                article_id: expect.any(Number),
+                topic: "cats",
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
+                comment_count: expect.any(Number),
+              })
+            );
+            expect(article.body).toBeUndefined();
+          });
+          });
+        });
+  test('200: serves an array of articles, filtered by topic and sorted by order queries', () => {
+      return request(app)
+        .get('/api/articles?sort_by=author&order=asc&topic=mitch')
+        .expect(200)
+        .then(({body}) => {
+          const { articles } = body;
+          expect(Array.isArray(articles)).toBe(true);
+          expect(articles).toBeSortedBy("author", { ascending: true });
+          articles.forEach((article) => {
+            expect(article).toEqual(
+              expect.objectContaining({
+                author: expect.any(String),
+                title: expect.any(String),
+                article_id: expect.any(Number),
+                topic: "mitch",
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
+                comment_count: expect.any(Number),
+              })
+            );
+            expect(article.body).toBeUndefined();
+          });
+          });
+        });
+  test('404: serves an error message if topic has no articles', () => {
+      return request(app)
+        .get('/api/articles?topic=paper')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('Topic not found');
+          });
+        });
```

Comment:
This should actually be a 200 status with an empty array as the result, because topic exists as a valid topic, but having no articles attached to it isnt invalid.

However if given an endpoint of `/api/articles?topic=banana` which doesnt exist, should actually give a 404 as this is invalid, but could exist as a topic